programs/test/selftest.c

```diff
@@ -76,7 +76,10 @@ static int calloc_self_test(int verbose)
     unsigned int buffer_4_size = 4097; /* Allocate more than the usual page size */
     unsigned char *buffer3 = mbedtls_calloc(buffer_3_size, 1);
     unsigned char *buffer4 = mbedtls_calloc(buffer_4_size, 1);
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Walloc-size-larger-than="
+    unsigned char *buffer5 = mbedtls_calloc(SIZE_MAX/2, SIZE_MAX/2);
```

Comment:
> I can see that the overly-large allocation code fails on older GCC's and windows builds - what approach do you prefer

The approach I'd prefer is to put this test in a test suite and have the parameters come from test data at runtime. Passing compile-time constants allows the compiler to optimize in a way that does not call the library function, making the test ineffective.

Reply:
Good idea, I'll add that.